### PR TITLE
Ability to delete old reports for given organization on demand

### LINKED ISSUE #35: Differ: ability to cleanup reports older that one week after weekly summary is send to given org_id users
Differ: ability to cleanup reports older that one week after weekly summary is send to given org_id users

### PR DESCRIPTION
**Pull request by tisnik**
# Description

Ability to delete old reports for given organization on demand

Fixes #35

## Type of change

- New feature (non-breaking change which adds functionality)

## Testing steps

N/A

## Checklist
* [x] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: differ/storage.go [hunk 1]
```diff
@@ -482,3 +509,57 @@ func (storage DBStorage) ReadLastNNotificationRecords(clusterEntry types.Cluster
 
 	return notificationRecords, nil
 }
+
+// CleanupForOrganization method deletes all reports older than specified
+// relative time for given organization ID.
+//
+// This method is to be used to cleanup older reports after weekly summary is
+// send.
```
Comment (epapbak) [inline differ/storage.go:517]: sent

File: differ/storage.go [hunk 2]
```diff
@@ -482,3 +509,57 @@ func (storage DBStorage) ReadLastNNotificationRecords(clusterEntry types.Cluster
...
+	}
+	return int(affected), nil
+}
+
+// CleanupNewReportsForOrganization method deletes all reports from
+// `new_reports` table older than specified relative time. Delete operation is
+// restricted for given organization ID.
+//
+// This method is to be used to cleanup older reports after weekly summary is
+// send.
```
Comment (epapbak) [inline differ/storage.go:546]: sent

**Review by epapbak — APPROVED** [review]

File: differ/storage.go (hunk 1 above)
Comment (tisnik) [reply differ/storage.go]: fixed

File: differ/storage.go (hunk 2 above)
Comment (tisnik) [reply differ/storage.go]: fixed

**Pull request merged**